--- a/plugin/group_replication/libmysqlgcs/src/bindings/xcom/gcs_xcom_networking.cc
+++ b/plugin/group_replication/libmysqlgcs/src/bindings/xcom/gcs_xcom_networking.cc
@@ -422,7 +422,6 @@
   */
 static bool sock_descriptor_to_string(int fd, std::string &out) {
   struct sockaddr_storage sa;
-  socklen_t addr_size = static_cast<socklen_t>(sizeof(struct sockaddr_storage));
   char saddr[INET6_ADDRSTRLEN];

   // get the sockaddr struct
@@ -431,7 +430,7 @@
   // try IPv4
   if (sa.ss_family == AF_INET) {
     if (inet_ntop(AF_INET, &(((struct sockaddr_in *)&sa)->sin_addr), saddr,
-                  addr_size)) {
+                  sizeof(saddr))) {
       out = saddr;
       return false;
     }
@@ -440,7 +439,7 @@
   // try IPv6
   if (sa.ss_family == AF_INET6) {
     if (inet_ntop(AF_INET6, &(((struct sockaddr_in6 *)&sa)->sin6_addr), saddr,
-                  addr_size)) {
+                  sizeof(saddr))) {
       out = saddr;
       return false;
     }
